src/xunit.performance.api/ScenarioExecutionResult.cs

```diff
@@ -1,23 +1,39 @@
-﻿using System.Collections.Generic;
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Xunit.Performance.Api.Profilers.Etw;
+using System.Collections.Generic;
 using System.Diagnostics;
 
 namespace Microsoft.Xunit.Performance.Api
 {
+    /// <summary>
+    /// Defines an interface for a performance scenario result.
+    /// </summary>
     public sealed class ScenarioExecutionResult
     {
-        public ScenarioExecutionResult(Process process)
+        /// <summary>
+        /// Initializes a new instance of the ScenarioExecutionResult class.
+        /// </summary>
+        /// <param name="process">Scenario benchmark process that was run.</param>
+        public ScenarioExecutionResult(System.Diagnostics.Process process)
         {
             ProcessExitInfo = new ProcessExitInfo(process);
         }
 
+        /// <summary>
+        /// State of the run scenario process.
+        /// </summary>
         public ProcessExitInfo ProcessExitInfo { get; }
 
+        /// <summary>
+        /// Binary .etl file name where the ETW traces were logged.
+        /// </summary>
         public string EventLogFileName { get; set; }
 
         /// <summary>
         /// This is the list of Pmc that were captured by the ETW event listener.
-        /// TODO: We should reconsider this type, as it is internal data and it does not contain the display name.
         /// </summary>
-        public IEnumerable<PerformanceMonitorCounter> PerformanceMonitorCounters { get; set; }
+        public ISet<PerformanceMonitorCounter> PerformanceMonitorCounters { get; set; }
```

Comment:
Based on our conversation, I'm trying to rationalize the model for returned results.  I think it is OK.  You basically have a set of processes that each have a set of modules.  Both of these objects contain the PMC data that was aggregated for each of them.  For example, each module has PMC data that showed up for that module.  Each process has PMC data for all of the modules that accrued PMC data in that process.  Is that right?

Reply:
Yes, that is correct.

---
In reply to: [146059975](https://github.com/Microsoft/xunit-performance/pull/242#discussion_r146059975) [](ancestors = 146059975)